utils/model.py

```diff
@@ -542,3 +587,21 @@ def parse_messages():
             yield "", "length"
         else:
             yield "", "stop"
+
+    def __general_model_stream(self, model: ModelTypes, tokenizer: AutoTokenizer, prompt: str,
+                               model_version: str, generation_config: GenerationConfig):
+        position = 0
+        start = 0
+        token_cnt = 0
+        input_tokens = tokenizer(prompt, return_tensors="pt")
+
+        model.generation_config.__dict__ = generation_config.__dict__
+        for response in model.generate_stream(**input_tokens.to(model.device), generation_config=generation_config):
+            token_cnt += 1
+            position = len(response)
+            yield response[start:position], None
+            start = position
+        if token_cnt == generation_config.__dict__['max_new_tokens']:
+            yield "", "length"
```

Comment:
interesting, the llama.cpp thinks that when stopped by `max_new_tokens` the stop reason is `stop`

https://github.com/ggerganov/llama.cpp/pull/7198